### PR TITLE
ENH: catch warnings in valid codepaths of KL functionals

### PR DESCRIPTION
**Pull request by kohr-h**
Small fix that gets rid of the obnoxious warnings in the KL functionals.

### CODE REVIEW

File: odl/solvers/functional/default_functionals.py [hunk 1]
```diff
@@ -1258,17 +1259,18 @@ def _call(self, x):
...
+        with np.errstate(invalid='ignore'):
+            if self.prior is None:
+                res = -(np.log(1 - x)).inner(self.domain.one())
+            else:
+                xlogy = scipy.special.xlogy(self.prior, 1 - x)
+                res = -self.domain.element(xlogy).inner(self.domain.one())
+
+        if not np.isfinite(res):
             # In this case, some element was larger than or equal to one
             return np.inf
```
Comment (adler-j) [inline odl/solvers/functional/default_functionals.py]: This has to be wrong, right? The gradient is a space element.

Comment (kohr-h) [reply odl/solvers/functional/default_functionals.py]: I see what you mean, issue of display :-) The folded part is all the way from `KullbackLeibler.gradient` to `KullbackLeiblerCrossEntropy._call`.
So this code is part of `KullbackLeiblerCrossEntropy._call`.

Comment (adler-j) [reply odl/solvers/functional/default_functionals.py]: Ok, then I'm good!

**Comment by kohr-h** [comment]
> Fix PEP and go!

Those are unfixable too long doc lines

    odl.solvers.nonsmooth.proximal_operators.proximal_convex_conj_kl_cross_entropy :

I'll leave them alone.

**Pull request merged**